### ISSUE TITLE
Shell script blocked if tooting with media

### ISSUE THREAD

**Issue by ValentinDrean** [ihabunek/toot#409]
Hi,

First of all thanks a lot to anyone working on toot cli, it is a super great & useful tool, also thanks to anyone reading this and sorry if this problem is more related to my instance or to my shell script, as I'm no expert & still learning. I struggled thinking where this would be better to ask & I told my self maybe solving the problem here would help others !

tldr; 
I have a shell script to post medias on file creation inside a folder with `inotifywait` for an account & it gives me a token access error, it does work for posting text only. I tried looking at `toot post --help` or `toot --help` & I couldn't find anything about giving a token again or something like that.

Script :
```
#!/bin/sh
MONITORDIR="$HOME/.local/share/TowerFall/TowerFall Replays/"
toot login --instance "https://piaille.fr"
inotifywait -m -r -e create --format '%w%f' "${MONITORDIR}" | while read NEWFILE
do
        toot post -m "${NEWFILE}" --debug 
done
```
Here is the output :
```
❯ ./monitor.sh
Looking up instance info...
Found instance Piaille running Mastodon version 4.2.0

This authentication method requires you to log into your Mastodon instance
in your browser, where you will be asked to authorize toot to access
your account. When you do, you will be given an authorization code
which you need to paste here.

This is the login URL:
https://piaille.fr/oauth/authorize/?response_type=code&redirect_uri=XXX&scope=read+write+follow&client_id=XXX

Open link in default browser? [Y/n]
Authorization code: XXX

Requesting access token...
Access token saved to config at: /home/roz/.config/toot/config.json

✓ Successfully logged in.
Setting up watches.  Beware: since -r was given, this may take a while!
Watches established.
Uploading media: /home/roz/.local/share/TowerFall/TowerFall Replays/replay0.gif
DEBUG:toot: --> POST https://piaille.fr/api/v2/media
DEBUG:toot: <-- POST https://piaille.fr/api/v2/media HTTP 202 643ms
Waiting for media to finish processing...
DEBUG:toot: --> GET https://piaille.fr/api/v1/media/111287631742795700
DEBUG:toot: <-- GET https://piaille.fr/api/v1/media/111287631742795700 HTTP 200 318ms
```

If we open [link sent from debug](https://piaille.fr/api/v1/media/111287625303029350) [or this one](https://piaille.fr/api/v1/media/111287631742795700) it says (in french) : `error | "Le jeton d’accès n’est pas valide"`, english translation attempt : `error | "access token not valid"`. I didn't contact piaille.fr instance yet.

Thanks a bunch ヽ(=^･ω･^=)丿

**Comment by thifranc** [comment]
Hello,

Je réponds en français mais est-ce que tu as bien vérifié que l'app que tu génères a les autorisations nécessaires pour write-medias ? de ce que tu décrire j'ai l'impression que non, mais j'ai un peu survolé l'issue et je n'ai pas testé le code 'too' donc je suis peut-être à côté de la plaque.

tldr english : maybe it's an issue with app-token access, meaning you'd be sure that it has the rights to upload medias.

![Screenshot_20231025_115517](https://github.com/ihabunek/toot/assets/17310666/aec84f8d-7dc9-4714-ae5b-7194bfe03b29)

**Comment by ValentinDrean** [comment]
Merci @thifranc !

J'ai déconnecté entre temps, j'ai pas réussi à trouver cette page que tu me montre à part en créant une Application avec Token, sauf que j'arrive pas à donner à manger à toot-cli avec !

I didn't find any documentation inside toot-cli to feed it with token & API id, anyone have an idea ? 

Thanks 🙏

**Comment by ValentinDrean** [comment]
Sorry for ghosting this issue again.

Apparently I had to put some text with my toot to make my script work with `--media` argument, I don't know why but in the end it worked. 

Example :

`toot post "toot" --media "${NEWFILE}" --description "toot"`

After that, I finally managed to create a automatic script to upload medias on Mastodon instance, [maybe it's interesting to be shared even if it's bugged](https://code.globenet.org/rozlav/auto-toot-media-from-folder) :shrug: 

Thanks to anyone reading or helping here !
